packages/google-pay-integration/src/google-pay-payment-processor.ts

```diff
@@ -63,6 +63,16 @@ export default class GooglePayPaymentProcessor {
         return this._gateway.getNonce(methodId);
     }
 
+    async getDeviceSessionId(): Promise<string | undefined> {
+        return this._gateway.getDeviceSessionId();
+    }
+
+    async getExtraData() {
+        return {
+            deviceSessionId: await this._gateway.getDeviceSessionId(),
+        };
+    }
```

Comment:
I don't really sure that this is a right way... getDeviceSessionId is relevant only for Braintree gateway, so I don't see any arguments why should we keep this method in the processor file.

```suggestion
    getExtraPaymentData() {
        return this._gateway.getExtraPaymentData();
    }
```

We as integration developers should be able to provide specific to our provider payment data (like formattedData for example), so the data should be build on gateway side, but not in the abstract class like this place.

Also, there should be an interface of the returned method's value.